airbyte-integrations/connectors/source-stripe/source_stripe/source.py

```diff
@@ -297,7 +297,15 @@ def streams(self, config: MutableMapping[str, Any]) -> List[Stream]:
             # The Refunds stream does not utilize the Events API as it created issues with data loss during the incremental syncs.
             # Therefore, we're using the regular API with the `created` cursor field. A bug has been filed with Stripe.
             # See more at https://github.com/airbytehq/oncall/issues/3090, https://github.com/airbytehq/oncall/issues/3428
-            CreatedCursorIncrementalStripeStream(name=_REFUND_STREAM_NAME, path="refunds", **incremental_args),
+            # Based on https://app.zenhub.com/workspaces/critical-connectors-659c693d906fcd00178900cf/issues/gh/airbytehq/airbyte-internal-issues/2450
+            # We are commenting this out just in case a rollback is needed. We are putting refunds back as UdatedCursorIncremental
+            # CreatedCursorIncrementalStripeStream(name=_REFUND_STREAM_NAME, path="refunds", **incremental_args),
+            UpdatedCursorIncrementalStripeStream(
+                name="refunds",
+                path="refunds",
+                event_types=["refund.created", "refund.updated", "charge.refunded", "charge.refund.updated"],
```

Comment:
It feels like there is a risk here that a single refund-related action would trigger two of these. For example, when one is created, getting both a `refund.created` and a `charge.refunded` or when a refund is updated, getting both a `refund.updated` and a `charge.refund.updated`.
This could then result in double-meeting the refund record. Have you verified the behavior and we only get one or other other?

Reply:
Not really. refund.created and refund.updated are only triggered when the Customer's cash balance is created. Whereas the charge.refunded and charge.refunded.updated are triggered when a refund attempt through a payment intent is generated. 
So, there's no duplication for those events, as they are triggered by different actions.